### PR TITLE
Add base network support

### PR DESCRIPTION
**Pull request by Tanya-atatakai**
- shorts are filtered based on the selected network
- NFT links contain scan URL of the short's chain (on both list and video pages)
- Additionally display short's chain network as discussed with @oaksprout 

Screenshot 2024-03-14 at 18 34 55

### CODE REVIEW

File: common-util/Contracts/index.jsx [hunk 1]
```diff
@@ -21,25 +21,43 @@ import {
...
+  AGENT_REGISTRY_ADDRESS_BASE,
+  AGENT_FACTORY_ADDRESS_BASE,
+  AGENT_MECH_ADDRESS_BASE,
+  AGENT_MULTISIG_ADDRESS_BASE,
+  BLOCKCHAIN_SHORTS_ADDRESS_BASE,
 } from 'common-util/AbiAndAddresses';
 import { getChainId, getProvider } from 'common-util/functions';
+import {
+  base, gnosis, neonMainnet, polygonZkEvm,
+} from 'viem/chains';
```
[the remark this answers is not in the document]
Comment (Tanya-atatakai) [reply common-util/Contracts/index.jsx]: fixed, thanks!

**Review by oaksprout — APPROVED** [review]

File: common-util/AbiAndAddresses/BlockchainShorts.jsx
Comment (truemiller) [inline common-util/AbiAndAddresses/BlockchainShorts.jsx:1]: Rename file to `blockchainShorts.js`, as it's not a component and will probably never contain JSX

File: common-util/AbiAndAddresses/agentFactoryX.jsx
Comment (truemiller) [inline common-util/AbiAndAddresses/agentFactoryX.jsx:1]: same, please make it .js

File: common-util/AbiAndAddresses/agentMechX.jsx
Comment (truemiller) [inline common-util/AbiAndAddresses/agentMechX.jsx:1]: same, please make it .js

File: common-util/AbiAndAddresses/agentRegistryX.jsx
Comment (truemiller) [inline common-util/AbiAndAddresses/agentRegistryX.jsx:1]: same, please make it .js

File: common-util/Contracts/index.jsx
Comment (truemiller) [inline common-util/Contracts/index.jsx:1]: same, please make it .js

**Review by truemiller — APPROVED** [review]
Small comments to resolve, around filename & extension.
Looks fine from a code perspective, but I can't access preview to check network logs, console errors .etc

Not related to this PR, but there are some things that could be improved:
- No need to store addresses and ABIs to together, seems like a pointless export, we can just define the addresses in Contracts/index or in a file specific to address constants
- We're using Web3, Viem, and Ethers. This is not appropriate, we should just use one. They all do the same thing. Arguably Viem is the most performant, but Ethers is fine and most commonly used in our applications, would advise moving away from Web3.js.

**Pull request merged**